### ISSUE TITLE
Vesta Data fixes: _LOC import / sync between trees (same wt-installation)

### ISSUE THREAD

**Issue by ungeahnt** [vesta-webtrees-2-custom-modules/vesta_shared_places#126]
Is there a possibility to import / sync the  _LOC entries  (incl. different names, languages, coordinates, notes, ...) from one tree to another one in a wt-installation?

'Vesta Data fixes' seems only to take data from 'wt geographic data' and GOV.

**Comment by ric2016** [comment]
No, there is no support for this other than what webtrees provides (via 'Merge Trees'), and there are no plans to support this.

It should be possible to merge just the _LOC records themselves from one GEDCOM file to another (outside webtrees), and then run one of the Vesta datafixes in order to map event places to the imported shared places.

**Comment by ungeahnt** [comment]
I've done some tests and I think there is actually no satisfying way to keep the _LOC-data between two or more pedigrees up-to-date.

The good news is, you can actually create a GEDCOM file which includes only _LOC data (+ GEDCOM header/footer) and you can import it into wt (e.g. via wt's merge function).

But... only if you are sure that no _LOC duplicates exists you can use wt's merge function without preliminary work (as we know, wt copies rather than merges). Otherwise you will have a lot of duplicates and due to the hierarchical LOC-structure, fixing is not quite easy.

Therefore, for the import/sync of _LOC data in wt's you have to
- create a GEDCOM file 
- merge/compare/edit manually the GEDCOM file (with an external editor)
- or fix the duplicates elaborately via wt's edit forms

I hope there will be a tool in the future with which a merge / synchronization is possible.  Of course, this does not have to be a Vesta module.
If _LOC becomes official in a future GEDCOM specification, then I hope that wt will support this and 'geograhic data'-db will become the central LOC management (incl. a import and synchronization function).

Is there already a statement from Greg about the integration of LOC in the wt core?

**Comment by ric2016** [comment]
Ultimately merging _LOCs from different trees (or other sources) is at least as complex as merging other kinds of records. I don't think it can ever be done automatically, and I'm not sure webtrees should even support this via additional functionality (there may be other more suitable tools for this).

If you plan to merge repeatedly - perhaps because you'd like to have a central place management? (or what is the actual use case here?) - I'd suggest to consider merging the trees altogether instead, or look for alternative solutions, such as linking to GOV-data.

> Is there already a statement from Greg about the integration of LOC in the wt core?

I don't think this is planned in the near future, if at all. Probably not until _LOC becomes an official part of GEDCOM, which in turn I'd be very suprised to see within the next couple of _years_.

**Comment by ungeahnt** [comment]
>  is at least as complex as merging other kinds of records ... I don't think it can ever be done automatically, 

never say never! ;) 
It's certainly not very simple, but you have a fixed structure (like language tags, the hierarchy, ... ), so at least a semi-automatic merge / data fix should be possible. I'll have a look if there isn't already something comparable. 

> perhaps because you'd like to have a central place management? (or what is the actual use case here?)

As soon as _LOC becomes more common, one has more and more the problem to synchronize LOC records. IMHO, it makes sense to have a central management (similar to webtrees 'geographic data').

Currently I have a public tree (with proven data) and 1-2 working trees. The latter contain imports/merges from other sources, which have to be checked and reworked. A central LOC management or an easy way to import LOC data would make the work much easier.

> which in turn I'd be very suprised to see within the next couple of years.
I hope it goes faster, but you're probably right.

> No, there is no support for this other than what webtrees provides (via 'Merge Trees'), and there are no plans to support this.
Since an enhancement is not planned, I close it
